packages/firestore/src/local/simple_db.ts

```diff
@@ -212,7 +212,23 @@ export class SimpleDb {
     return Number(version);
   }
 
-  constructor(private db: IDBDatabase) {}
+  private versionChangeListener: (
+    event: IDBVersionChangeEvent
+  ) => Promise<void>;
+
+  constructor(private db: IDBDatabase) {
+    this.db.onversionchange = (event: IDBVersionChangeEvent) => {
+      return this.versionChangeListener(event);
+    };
+  }
+
+  setVersionChangeListener(
+    versionChangeListener: (event: IDBVersionChangeEvent) => Promise<void>
+  ): void {
+    this.versionChangeListener = async event => {
```

Comment:
Do we need to manage our own `versionChangeListener` or could we just have this setter assign to `this.db.onversionchange` directly?

Or does this added indirection help out somehow?